snapcraft/cli/_errors.py

```diff
@@ -53,40 +67,52 @@ def exception_handler(exception_type, exception, exception_traceback, *,
 
     exit_code = 1
     is_snapcraft_error = issubclass(exception_type, errors.SnapcraftError)
-
-    if debug:
+    is_raven_setup = RavenClient is not None
+
+    # The use cases:
+    # - a non snapcraft handled error occurs and raven is setup,
+    #   so we go over confirmation logic showing the relevant traceback
+    # - a non snapcraft handled error occurs and raven is not setup,
+    #   so we just show the traceback
+    # - a snapcraft handled error occurs, --debug is set so a traceback
+    #   is shown
+    # - a snapcraft handled error occurs, --debug not set so only the
+    #   exception message is shown
+    if is_raven_setup and not is_snapcraft_error:
+        is_env_send_data = os.environ.get(
+            'SNAPCRAFT_SEND_ERROR_DATA', 'n') == 'y'
+        msg = _MSG_SEND_TO_SENTRY_TRACEBACK_CONFIRM
+        click.echo(_MSG_SEND_TO_SENTRY_TRACEBACK)
         traceback.print_exception(
             exception_type, exception, exception_traceback)
-
-    if not is_snapcraft_error:
-        echo.error('Sorry, Snapcraft had an internal error.')
-        if _is_send_error_data():
+        if is_env_send_data or click.confirm(msg):
+            if is_env_send_data:
+                click.echo(_MSG_SEND_TO_SENTRY_ENV)
             _submit_trace(exception)
-
-    should_print_error = not debug and (
-        exception_type != errors.ContainerSnapcraftCmdError)
```

Comment:
Should we still be accounting for `ContainerSnapcraftCmdError`?

Reply:
it inherits from `SnapcraftError`. I am not sure about this logic anymore.